### PR TITLE
[7495] Entering a number from keyboard in Publish - Schedule date is causing the dialog to shown an error.

### PR DESCRIPTION
**Pull request by jvega190**
https://github.com/craftercms/craftercms/issues/7495

### CODE REVIEW

File: ui/app/src/components/DateTimePicker/DateTimePicker.tsx [hunk 1]
```diff
@@ -264,16 +266,7 @@ function DateTimePicker(props: DateTimePickerProps) {
...
                     },
                 inputProps: {
-                  onChange: handlePopupOnlyInputChange,
-                  value: asLocalizedDateTime(internalDate, localeCode, {
-                    hour12,
-                    hour: dateTimeFormatOptions?.hour || '2-digit',
-                    minute: dateTimeFormatOptions?.minute || '2-digit',
-                    // If the timezone control isn't displayed, the time displayed may
-                    // be misleading/unexpected to the user, so if timezone isn't displayed,
-                    // display timezone here.
```
Comment (rart) [inline ui/app/src/components/DateTimePicker/DateTimePicker.tsx:274]: We need to address this somehow.

**Review by rart — CHANGES_REQUESTED** [review]

File: ui/app/src/components/DateTimePicker/DateTimePicker.tsx (hunk 1 above)
Comment (jvega190) [reply ui/app/src/components/DateTimePicker/DateTimePicker.tsx:274]: what if we go back to setting the value with the `asLocalizedDateTime` but with the default format values (`2-digit` for hour and minute), and that way we keep the displaying of the timezone when needed?

btw, the showTimeZoneSelector prop is not currently being set to false in any usage

Comment (jvega190) [reply ui/app/src/components/DateTimePicker/DateTimePicker.tsx:274]: I pushed the change I mentioned in the comment above, please let me know if not what we want so I can revert.

**Review by rart — CHANGES_REQUESTED** [review]
You didn't test the `!showTimeZoneSelector` case... The keyboard doesn't work well.

**Review by rart — APPROVED** [review]

**Pull request merged**